runtime/src/main/cpp/Arrays.cpp

```diff
@@ -131,6 +131,102 @@ KInt Kotlin_ByteArray_getArrayLength(KConstRef thiz) {
+}
+
+KShort Kotlin_ByteArray_getShortAt(KConstRef thiz, KInt index) {
+  const ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 1) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  return *reinterpret_cast<const KShort*>(ByteArrayAddressOfElementAt(array, index));
+}
+
+KInt Kotlin_ByteArray_getIntAt(KConstRef thiz, KInt index) {
+  const ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 3) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  return *reinterpret_cast<const KInt*>(ByteArrayAddressOfElementAt(array, index));
+}
+
+KLong Kotlin_ByteArray_getLongAt(KConstRef thiz, KInt index) {
+  const ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 7) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  return *reinterpret_cast<const KLong*>(ByteArrayAddressOfElementAt(array, index));
+}
+
+KFloat Kotlin_ByteArray_getFloatAt(KConstRef thiz, KInt index) {
+  const ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 3) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  return *reinterpret_cast<const KFloat*>(ByteArrayAddressOfElementAt(array, index));
+}
+
+KDouble Kotlin_ByteArray_getDoubleAt(KConstRef thiz, KInt index) {
+  const ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 7) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  return *reinterpret_cast<const KDouble*>(ByteArrayAddressOfElementAt(array, index));
+}
+
+void Kotlin_ByteArray_setCharAt(KRef thiz, KInt index, KChar value) {
+  ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 1) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  *reinterpret_cast<KChar*>(ByteArrayAddressOfElementAt(array, index)) = value;
+}
+
+void Kotlin_ByteArray_setShortAt(KRef thiz, KInt index, KShort value) {
+  ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 1) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  *reinterpret_cast<KShort*>(ByteArrayAddressOfElementAt(array, index)) = value;
+}
+
+void Kotlin_ByteArray_setIntAt(KRef thiz, KInt index, KInt value) {
+  ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 3) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  *reinterpret_cast<KInt*>(ByteArrayAddressOfElementAt(array, index)) = value;
+}
+
+void Kotlin_ByteArray_setLongAt(KRef thiz, KInt index, KLong value) {
+  ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 7) >= array->count_) {
+    ThrowArrayIndexOutOfBoundsException();
+  }
+  *reinterpret_cast<KLong*>(ByteArrayAddressOfElementAt(array, index)) = value;
+}
+
+void Kotlin_ByteArray_setFloatAt(KRef thiz, KInt index, KChar value) {
+  ArrayHeader* array = thiz->array();
+  if (static_cast<uint32_t>(index + 3) >= array->count_) {
```

Comment:
Have you checked that conversion is correct in case of INFINITY, or other huge floating-point value?

The C++ standard says that such conversions are Undefined Behaviors

Reply:
Guess here correct is any behavior that is fast :).